loopy/transform/callable.py

```diff
@@ -157,7 +158,8 @@ def map_subscript(self, expr, expn_state):
                 flatten_index -= (dim_tag.stride * ind)
                 new_indices.append(ind)
 
-            new_indices = tuple(simplify_via_aff(i) for i in new_indices)
+            new_indices = tuple(simplify_using_aff(
+                self.callee_knl, i) for i in new_indices)
```

Comment:
Because of this default behaviour, one tests `test_simplify_indices` fails. Should I be doing this by default or should the user do it?

Reply:
Getting rid of [these lines](https://github.com/inducer/loopy/blob/4c9fb2b786aafc006cd28243517d578ffbaa5983/test/test_transform.py#L1297-L1299) in the test should be fine.

Reply:
@inducer, since we are calling `simplify_using_aff` here, the inlining produces a kernel that is invariant under `simplify_indices` call. Therefore I used a different test for `simplify_indices`